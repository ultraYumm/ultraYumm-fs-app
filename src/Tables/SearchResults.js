import React, { Component } from "react";
import "../Font/Font.css";
import SaveButton from "../FormElements/SaveButton";
import ResetButton from "../FormElements/ResetButton";
import PrintButton from "../FormElements/PrintButton";
import AddCustomButton from "../FormElements/AddCustomButton";
import SearchMoreButton from "../FormElements/SearchMoreButton";
import "./Tables.css";
import { NavLink } from "react-router-dom";
import TableItemImage from "../ItemDetails/TableItemImage";
import ItemName from "../ItemDetails/ItemName";
import ItemBrand from "../ItemDetails/ItemBrand";
import ServingQuant from "../ItemDetails/ServingQuant";
import ServingUnit from "../ItemDetails/ServingUnit";
import CalsPerHg from "../ItemDetails/CalsPerHg";
import CalsPerServing from "../ItemDetails/CalsPerServing";
import ServingWeight from "../ItemDetails/ServingWeight";
import Add from "../ItemDetails/Add";
import { DEFAULTITEM } from "../Defaults";
import ImageHeader from "./ImageHeader"
import ItemHeader from "./ItemHeader"
import BrandHeader from "./BrandHeader";
import ServQuantHeader from "./ServQuantHeader";
import UnitHeader from "./UnitHeader";
import WeightGHeader from "./WeightGHeader";
import TotalCalHeader from "./TotalCalHeader";
import TotalCalsPhgHeader from "./TotalCalsPhgHeader";
import AddHeader from "./AddHeader";
import BackButton from '../FormElements/BackButton';

class SearchResults extends Component {
  static defaultProps = {
    results: {},
    common: [],
    branded: [],
    item:  {DEFAULTITEM},
    thumb: "",
  };

  constructor(props) {
    super(props);
    this.state = {
      serving_qty: 0,
      id: ""
    
      
    };
  }

  adjustQuant = (input, id) => {
    
    this.setState({
      serving_qty: input,
      id: id
    });
  };



  render() {

    const common = this.props.common;
    const branded = this.props.branded;
    const trips = this.props.trips;
    console.log(trips)
    Array.prototype.push.apply(common, branded);

    const commonNutrients = () => {
      var results = common.map((item) => item.full_nutrients);
      return results;
    };

    const nutrients = commonNutrients();

    const commonCalsPerS = () => {
      var results = nutrients.map((nutrient) => {
        var data = nutrient.filter((item) => item.attr_id == 208);
        return data;
      });
      return results;
    };

    const calorieArray = commonCalsPerS();

    const removeNoCalNutrients = () => {
      for (let i = 0; i < common.length; i++) {
        common[i].full_nutrients = calorieArray[i];
        
        common[i].image = common[i].photo.thumb
        const replace = "%";
        common [i].calsPerServing = common[i].full_nutrients[0].value
        common[i].id = common[i].food_name.replace(replace, "");
        common[i].calsPhg = !common[i].serving_weight_grams
          ? Math.round(common[i].full_nutrients[0].value)
          : Math.round(
              (common[i].full_nutrients[0].value /
                common[i].serving_weight_grams) *
                100
            );
      }
      return common;
    };

    const newBrandCommonArr = removeNoCalNutrients();

    newBrandCommonArr.sort(
      (b, a) => parseFloat(a.calsPhg) - parseFloat(b.calsPhg)
    );

    console.log(newBrandCommonArr)

    

    
    return (
      <section>
        <BackButton/>        
        <div>
          <h2 className="montebello searchResultsTitle white">
            {" "}
            See your highest energy results!
          </h2>
        </div>

        <div className="filterButtonContainer tableAdjust">
          <SaveButton />
          <PrintButton />
          <ResetButton />
        </div>
        
          <div className="filterButtonContainer moreContainer">
            <NavLink to={`/add-custom`}
           >
              <AddCustomButton />
            </NavLink>

            <NavLink to={`/`}>
              <SearchMoreButton />
            </NavLink>
          </div>
          <table id="search-results" className="primaryFont">
            <tbody>
              <tr className="blueBackground white">
                <ImageHeader/>
                <ItemHeader/>
                <BrandHeader/>
                <ServQuantHeader/>
                <UnitHeader/>
                <WeightGHeader/>
                <TotalCalHeader/> 
                <TotalCalsPhgHeader/> 
                <AddHeader/>
              </tr>

              {newBrandCommonArr.map((item, key) => (
                <tr className="one whiteBackground black" key={key}>
               
                    <TableItemImage image = {item.image}/>
                 
                  <td className="itemH">
                    <NavLink
                     className = "noDeco"
                      to={`/item/${item.id}`}
                      onClick={() => {
                        const selectedItem = item;
                        this.props.handleSelectedItem(selectedItem);
                      }}
                    >
                      <ItemName 
                      name = {item.food_name} 
                      item = {item}
                      handleSelectedItem = {this.props.handleSelectedItem}/>
                    </NavLink>
                  </td>

                    <ItemBrand
                    brand = {!item.brand_name ? "common" : item.brand_name}
                    />
                    <ServingQuant 
                    quant = {Math.round(item.serving_qty)}
                    id = {item.id}
                    handleAdjustQuant = {(inputValue, id) =>
                      this.adjustQuant(inputValue, id)
                    }/>
                    <ServingUnit unit={item.serving_unit} />
                    <ServingWeight
                    weight =  {Math.round(item.serving_weight_grams)}
                    result = {Math.round(this.state.serving_qty * item.serving_weight_grams).toFixed(0)
                      .replace(/(\d)(?=(\d{3})+(?!\d))/g, "$1,")}
                    />
                    <CalsPerServing
                    calories =  {Math.round(item.calsPerServing)}
                    result = {Math.round(this.state.serving_qty * item.calsPerServing ).toFixed(0)
                      .replace(/(\d)(?=(\d{3})+(?!\d))/g, "$1,")}
                    />   
                    <CalsPerHg
                      calsPHg = {item.calsPhg}
                    />
                  <Add 
                  trips={trips} />
                                   </tr>
              ))}
            </tbody>
          </table>
        
      </section>
    );
  }
}

export default SearchResults;

