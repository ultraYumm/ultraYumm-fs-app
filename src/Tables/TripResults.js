import React, { Component } from "react";
import "../Font/Font.css";
import TripName from "../TripDetails/TripName";
import TripDates from "../TripDetails/TripDates";
import ItemTypes from "../CustomItem/ItemTypes";
import TripTravelers from "../TripDetails/TripTravelers";
import TripYear from "../TripDetails/TripYear";
import Moment from "react-moment";
import SaveButton from "../FormElements/SaveButton";
import ResetButton from "../FormElements/ResetButton";
import PrintButton from "../FormElements/PrintButton";
import AddCustomButton from "../FormElements/AddCustomButton";
import GoToFiltersButton from "../FormElements/GoToFiltersButton";
import SearchMoreButton from "../FormElements/SearchMoreButton";
import TripPackGraph from "../TripDetails/TripPackGraph";
import TripCalorieGraph from "../TripDetails/TripCalorieGraph";
import { NavLink } from "react-router-dom";
import ItemName from "../ItemDetails/ItemName";
import ServingQuant from "../ItemDetails/ServingQuant";
import ServingUnit from "../ItemDetails/ServingUnit";
import ServingWeight from "../ItemDetails/ServingWeight";
import "./Tables.css";
import CalsPerServing from "../ItemDetails/CalsPerServing";
import Delete from "../ItemDetails/Delete";
import ItemHeader from "./ItemHeader"
import ServQuantHeader from "./ServQuantHeader";
import UnitHeader from "./UnitHeader";
import WeightGHeader from "./WeightGHeader";
import TotalCalHeader from "./TotalCalHeader";
import DateHeader from "./DateHeader";
import TypeHeader from "./TypeHeader";
import TravlrHeader from "./TravlrHeader";
import DeleteHeader from "./DeleteHeader";
import BackButton from '../FormElements/BackButton';


class TripResults extends Component {
  static defaultProps = {
    selectedTrip: [],
    
  };

  render() {
    const selectedTrip = this.props.selectedTrip;
    
    console.log(selectedTrip)
    const selectedTripItems = this.props.selectedTripItems;
    const tripItems = this.props.tripItems;
    const imageArray = tripItems.map((items) => items.image);
    const food_nameArray = tripItems.map((items) => items.food_name);
    const brandArray = tripItems.map((items) => items.brand_name);
    const fixedUnitArray = tripItems.map((items) => items.serving_unit);
    const fixedCalPerUnitArray = tripItems.map((items) => items.calsPerServing);
    const selectServingQuantArray = selectedTripItems
      .map((items) => items.selected_serving_qty)
      .sort();
    const fixedServingWeightArray = tripItems.map(
      (items) => items.serving_weight_grams
    );
    const itemTypes = this.props.itemTypes;
    const results = selectedTripItems.map((item, i) => {
    const food_name = food_nameArray[i];
    const image = imageArray[i];
    const brand_name = brandArray[i];
    const fixedUnit = fixedUnitArray[i];
    const calsPerServing = fixedCalPerUnitArray[i];
    const serving_weight_grams = fixedServingWeightArray[i];
    const resultsObject = {
        image,
        food_name,
        brand_name,
        fixedUnit,
        serving_weight_grams,
        calsPerServing,
        ...item,
      };

      return resultsObject;
    });
    const reducer = (accumulator, currentValue) => accumulator + currentValue;
    const sumCalsPerServing = () => {
      if (fixedCalPerUnitArray.length === 0) {
        const defaultResult = 1;
        return defaultResult;
      }
      return fixedCalPerUnitArray.reduce(reducer);
    };
    const sumWeightsPerServing = () => {
      if (fixedServingWeightArray.length === 0) {
        const defaultResult = 1;
        return defaultResult;
      }
      return fixedServingWeightArray.reduce(reducer);
    };
    console.log(sumWeightsPerServing());
    const sumServings = () => {
      if (selectServingQuantArray.length === 0) {
        const defaultResult = 1;
        return defaultResult;
      }

      return selectServingQuantArray.reduce(reducer);
    };

    const totalCals = sumCalsPerServing() * sumServings();
    const totalWeightKg = (sumWeightsPerServing() * sumServings()) / 1000;

    return (
      <section className="lightBlueBackground">
        <div className="charts redBackground">
         
          <TripCalorieGraph cals={totalCals}/>
          <TripPackGraph weight={totalWeightKg}/>
        </div>
        <div>
          <h2 className="montebello">
            <span className="resultsTitle montebello white">
              <TripName selectedTrip={selectedTrip}/>Results!
            </span>
            <span className="resultsTitle">
              <TripYear selectedTrip={selectedTrip}/>
            </span>
          </h2>
        </div>
        <form>
          <div className="filterButtonContainer">
          <BackButton/>
            <SaveButton/>
            <PrintButton/>
            <ResetButton/>
          </div>

          <div className="filterButtonContainer moreContainer">
            <NavLink to={`/trip-filter/${selectedTrip[0].food_name}`}>
              <GoToFiltersButton />
            </NavLink>

            <NavLink to={`/add-custom/${selectedTrip[0].food_name}`}>
              <AddCustomButton />
            </NavLink>

            <NavLink to={`/`}>
              <SearchMoreButton />
            </NavLink>
          </div>

          <table id="results-filtered" className="primaryFont">
            <tbody>
              <tr className="blueBackground white">
              
                <DateHeader/>
                <TypeHeader/>
                <TravlrHeader/>
                <ItemHeader/>
                <ServQuantHeader/>
                <UnitHeader/>
                <WeightGHeader/>
                <TotalCalHeader/>
                <DeleteHeader/>
               
              </tr>

              {results.map((item, key) => (
                <tr className="one whiteBackground black" key={item.itemId}>
                

                  <td className="date">
                    <Moment format="ddd-MMM-DD">{item.tripDay}</Moment>
                    <div className="tableScroll tableAdjust add">
                      {" "}
                      <TripDates selectedTrip={selectedTrip}/>
                    </div>
                  </td>

                  <td className="type">
                    {item.type}
                    <div className="tableScroll tableAdjust add">
                      <ItemTypes itemTypes={itemTypes}/>
                    </div>
                  </td>

                  <td className="traveler">
                    {item.travName}
                    <div className="tableScroll tableAdjust add">
                      <TripTravelers
                        selectedTrip={selectedTrip}
                      />
                    </div>
                  </td>

                  <td className="itemH">
                  <NavLink
                  className = "noDeco"
                      to={`/trip-item/${item.itemId}`}

                    onClick={() => {
                    const selectTripItem =  item
                    this.props.handleSelectTripItem(selectTripItem)}}                     
                       >
                      <ItemName 
                      name = {item.food_name} />
                      </NavLink>
                    </td>

                 

                 
                  <ServingQuant 
                    quant = {Math.round(item.selected_serving_qty)} />
                  
                 
                  <ServingUnit unit={item.fixedUnit}/>
                 

                  <ServingWeight
                   weight = {item.selected_serving_qty * item.serving_weight_grams}/>

                  <CalsPerServing
                    calories =
                    {(item.selected_serving_qty * item.calsPerServing)
                      .toFixed(0)
                      .replace(/(\d)(?=(\d{3})+(?!\d))/g, "$1,")}/>
                  <Delete/>
                 
                </tr>
              ))}
            </tbody>
          </table>
        </form>
      </section>
    );
  }
}

export default TripResults;

