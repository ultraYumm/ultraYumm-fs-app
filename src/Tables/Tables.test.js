import React from 'react';
import ReactDOM from 'react-dom';
import TripResults from './TripResults';
import SearchResults from './SearchResults';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';



it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <BrowserRouter>
     <TripResults />
  </BrowserRouter> 
  , div);
  ReactDOM.unmountComponentAtNode(div);
});


it('renders the UI as expected', () => {
  const tree = renderer
    .create( <BrowserRouter>
      <TripResults />
       </BrowserRouter> )
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });


it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    
      <BrowserRouter>
       <SearchResults/>
      </BrowserRouter>,
    div);
    ReactDOM.unmountComponentAtNode(div);
  });


it('renders the UI as expected', () => {
  const tree = renderer
    .create( <BrowserRouter>
        <SearchResults/>
       </BrowserRouter> )
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });
