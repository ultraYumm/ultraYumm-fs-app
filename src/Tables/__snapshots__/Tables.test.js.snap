// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="lightBlueBackground"
>
  <div
    className="lightBlueBackground sticky"
  >
    <div
      className="charts sticky cloudBlueBackground"
    >
      <div
        className="primaryFont black"
      >
         
        <span
          className="graphText"
        >
          149
           cals 
        </span>
        <i
          className="fas fa-fire-alt"
        />
      </div>
      <div
        className="primaryFont black"
      >
        <span
          className="graphText"
        >
          12
           kg 
        </span>
        <i
          className="fas fa-hiking"
        />
      </div>
    </div>
    <div
      className="iconButtonContainer sticky"
    >
      <i
        className="fas fa-arrow-circle-left skin arrowButtons"
        onClick={[Function]}
      />
      <i
        className="fas fa-arrow-circle-right skin forward arrowButtons"
        onClick={[Function]}
      />
    </div>
    <h2
      className="montebello black sticky resultsTitle"
    >
      <div
        className=" montebello black sticky resultsTitle"
      >
        <div>
          My awesome trip
        </div>
         Results! 
        <div
          className="montebello"
        >
          
        </div>
      </div>
    </h2>
    <div
      className="iconButtonContainer"
    >
      <button
        className="tooltip iconButtons"
      >
        <i
          className="fas fa-paw black"
          onClick={[Function]}
          type="submit"
        >
          <span
            className="tooltiptext primaryFont white"
          >
            print
          </span>
        </i>
      </button>
      <div
        className="primaryFont white redBackground moreButton sticky"
        onClick={[Function]}
      >
        delete selected
        <span
          className="primaryFont text"
        >
          : 
          
        </span>
      </div>
    </div>
    <div
      className="filterButtonContainer moreContainer sticky"
    >
      <a
        aria-current={null}
        href="/add-custom/My favorite treat"
        onClick={[Function]}
      >
        <div
          className="montebello red skinBackground moreButton sticky"
        >
          Make your own item
        </div>
      </a>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <div
          className="montebello red skinBackground moreButton sticky"
        >
          Search more items
        </div>
      </a>
    </div>
  </div>
  <table
    className="primaryFont desk"
    id="results-filtered"
  >
    <tbody>
      <tr
        className="blueBackground white"
      >
        <th
          className="date"
          id="date header"
        >
          <span
            className="hidden"
          >
            Date
          </span>
          <i
            alt="date header"
            className="fas fa-calendar-day"
          />
        </th>
        <th
          className="type"
          id="item type (ie breakfast) header"
        >
          <span
            className="hidden"
          >
            Type
          </span>
          <i
            alt="item type (ie breakfast) header"
            className="fas fa-utensils"
          />
        </th>
        <th
          className="traveler"
          id="traveler name header"
        >
          <span
            className="hidden"
          >
            Traveler name
          </span>
          <i
            alt="traveler name header"
            className="fas fa-user-circle"
          />
        </th>
        <th
          className="itemH"
          id="item name header"
        >
          <span
            className="hidden"
          >
            Name
          </span>
          <i
            alt="item name header"
            className="fas fa-carrot"
          />
        </th>
        <th
          className="servingH"
          id="serving quantity header"
        >
          <span
            className="hidden"
          >
            Quantity
          </span>
          <i
            alt="serving quantity header"
            className="fas fa-coins"
          />
        </th>
        <th
          className="unitH"
          id="unit ie tablespoon or cup header"
        >
          <span
            className="hidden"
          >
            Serving unit
          </span>
          <i
            alt="unit ie tablespoon or cup header"
            className="fas fa-ruler-vertical"
          />
        </th>
        <th
          className="weightH gram"
          id="weight in grams header"
        >
          <span
            className="hidden"
          >
            Serving weight
          </span>
          <i
            alt="weight in grams header"
            className="fas fa-balance-scale"
          />
        </th>
        <th
          className="caloriesH cal calS"
          id="total calories header"
        >
          <span
            className="hidden"
          >
            Total calories
          </span>
          <i
            alt="total calories header"
            className="fas fa-fire-alt"
          />
        </th>
        <th
          className="deleteH"
          id="delete header"
        >
          <span
            className="hidden"
          >
            Delete
          </span>
          <i
            alt="delete header"
            className="fas fa-trash-alt"
          />
        </th>
      </tr>
      <tr
        className="one whiteBackground black"
      >
        <td
          className="date"
        >
          <time
            dateTime={"2020-07-06T09:24:42.474Z"}
          >
            Mon-Jul-06
          </time>
        </td>
        <td
          className="type"
        />
        <td
          className="traveler"
        />
        <td
          className="itemH"
        >
          <a
            aria-current={null}
            className="noDeco"
            href="/trip-item/undefined"
            onClick={[Function]}
          >
            <div
              className="purple"
            >
              My favorite treat
            </div>
          </a>
        </td>
        <td
          className="servingH black"
          headers="serving quantity header"
        >
          1
        </td>
        <td
          className="unitH"
          headers="unit ie tablespoon or cup header"
        >
          a handful
        </td>
        <td
          className="weightH  gram"
          headers="weight in grams header"
        >
          100
           total gms
        </td>
        <td
          className="weightH  gram"
        >
          120
           total cals
        </td>
        <td
          className="delete"
        >
           
          <label
            className="tooltiptext delete"
          >
            click and confirm
            <input
              onClick={[Function]}
              type="radio"
            />
          </label>
        </td>
      </tr>
      <tr
        className="one whiteBackground black"
      >
        <td
          className="date"
        >
          <time
            dateTime={"2020-07-06T09:24:42.498Z"}
          >
            Mon-Jul-06
          </time>
        </td>
        <td
          className="type"
        />
        <td
          className="traveler"
        />
        <td
          className="itemH"
        >
          <a
            aria-current={null}
            className="noDeco"
            href="/trip-item/undefined"
            onClick={[Function]}
          >
            <div
              className="purple"
            >
              My favorite beverage
            </div>
          </a>
        </td>
        <td
          className="servingH black"
          headers="serving quantity header"
        >
          1
        </td>
        <td
          className="unitH"
          headers="unit ie tablespoon or cup header"
        >
          oz
        </td>
        <td
          className="weightH  gram"
          headers="weight in grams header"
        >
          2
           total gms
        </td>
        <td
          className="weightH  gram"
        >
          28
           total cals
        </td>
        <td
          className="delete"
        >
           
          <label
            className="tooltiptext delete"
          >
            click and confirm
            <input
              onClick={[Function]}
              type="radio"
            />
          </label>
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="primaryFont whiteBackground mobileOnly"
    id="results-filtered"
  >
    <tbody>
      <tr
        className="mobile"
      >
        <td
          className="noBorder"
        >
          <img
            alt=""
            className="tableImage "
            src="https://d2eawub7utcl6.cloudfront.net/images/nix-apple-grey.png"
          />
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr
        className="itemH"
      >
        <th
          className="itemH"
          id="item name header"
        >
          <span
            className="hidden"
          >
            Name
          </span>
          <i
            alt="item name header"
            className="fas fa-carrot"
          />
        </th>
        <td>
          <a
            aria-current={null}
            className="noDeco"
            href="/trip-item/undefined"
            onClick={[Function]}
          >
            <div
              className="purple"
            >
              My favorite treat
            </div>
          </a>
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="brandH"
          id="brand header"
        >
          <span
            className="hidden"
          >
            Brand
          </span>
          <i
            alt="brand header"
            className="fas fa-tag"
          />
        </th>
        <td
          className="brandId"
          headers="brand header"
        >
          common
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="date"
          id="date header"
        >
          <span
            className="hidden"
          >
            Date
          </span>
          <i
            alt="date header"
            className="fas fa-calendar-day"
          />
        </th>
        <td
          className="date"
        >
          <time
            dateTime={"2020-07-06T09:24:42.500Z"}
          >
            Mon-Jul-06
          </time>
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="type"
          id="item type (ie breakfast) header"
        >
          <span
            className="hidden"
          >
            Type
          </span>
          <i
            alt="item type (ie breakfast) header"
            className="fas fa-utensils"
          />
        </th>
        <td
          className="type"
        />
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="traveler"
          id="traveler name header"
        >
          <span
            className="hidden"
          >
            Traveler name
          </span>
          <i
            alt="traveler name header"
            className="fas fa-user-circle"
          />
        </th>
        <td
          className="type"
        />
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="unitH"
          id="unit ie tablespoon or cup header"
        >
          <span
            className="hidden"
          >
            Serving unit
          </span>
          <i
            alt="unit ie tablespoon or cup header"
            className="fas fa-ruler-vertical"
          />
        </th>
        <td
          className="unitH"
          headers="unit ie tablespoon or cup header"
        >
          a handful
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="servingH"
          id="serving quantity header"
        >
          <span
            className="hidden"
          >
            Quantity
          </span>
          <i
            alt="serving quantity header"
            className="fas fa-coins"
          />
        </th>
        <td
          className="servingH black"
          headers="serving quantity header"
        >
          1
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="weightH gram"
          id="weight in grams header"
        >
          <span
            className="hidden"
          >
            Serving weight
          </span>
          <i
            alt="weight in grams header"
            className="fas fa-balance-scale"
          />
        </th>
        <td
          className="weightH  gram"
          headers="weight in grams header"
        >
          100
           total gms
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="caloriesH cal calS"
          id="total calories header"
        >
          <span
            className="hidden"
          >
            Total calories
          </span>
          <i
            alt="total calories header"
            className="fas fa-fire-alt"
          />
        </th>
        <td
          className="weightH  gram"
        >
          120
           total cals
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="deleteH"
          id="delete header"
        >
          <span
            className="hidden"
          >
            Delete
          </span>
          <i
            alt="delete header"
            className="fas fa-trash-alt"
          />
        </th>
        <td
          className="delete"
        >
           
          <label
            className="tooltiptext delete"
          >
            click and confirm
            <input
              onClick={[Function]}
              type="radio"
            />
          </label>
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="primaryFont whiteBackground mobileOnly"
    id="results-filtered"
  >
    <tbody>
      <tr
        className="mobile"
      >
        <td
          className="noBorder"
        >
          <img
            alt=""
            className="tableImage "
            src="https://nutritionix-api.s3.amazonaws.com/5572031b36869fb5124af830.jpeg"
          />
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr
        className="itemH"
      >
        <th
          className="itemH"
          id="item name header"
        >
          <span
            className="hidden"
          >
            Name
          </span>
          <i
            alt="item name header"
            className="fas fa-carrot"
          />
        </th>
        <td>
          <a
            aria-current={null}
            className="noDeco"
            href="/trip-item/undefined"
            onClick={[Function]}
          >
            <div
              className="purple"
            >
              My favorite beverage
            </div>
          </a>
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="brandH"
          id="brand header"
        >
          <span
            className="hidden"
          >
            Brand
          </span>
          <i
            alt="brand header"
            className="fas fa-tag"
          />
        </th>
        <td
          className="brandId"
          headers="brand header"
        >
          My special instant mix
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="date"
          id="date header"
        >
          <span
            className="hidden"
          >
            Date
          </span>
          <i
            alt="date header"
            className="fas fa-calendar-day"
          />
        </th>
        <td
          className="date"
        >
          <time
            dateTime={"2020-07-06T09:24:42.502Z"}
          >
            Mon-Jul-06
          </time>
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="type"
          id="item type (ie breakfast) header"
        >
          <span
            className="hidden"
          >
            Type
          </span>
          <i
            alt="item type (ie breakfast) header"
            className="fas fa-utensils"
          />
        </th>
        <td
          className="type"
        />
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="traveler"
          id="traveler name header"
        >
          <span
            className="hidden"
          >
            Traveler name
          </span>
          <i
            alt="traveler name header"
            className="fas fa-user-circle"
          />
        </th>
        <td
          className="type"
        />
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="unitH"
          id="unit ie tablespoon or cup header"
        >
          <span
            className="hidden"
          >
            Serving unit
          </span>
          <i
            alt="unit ie tablespoon or cup header"
            className="fas fa-ruler-vertical"
          />
        </th>
        <td
          className="unitH"
          headers="unit ie tablespoon or cup header"
        >
          oz
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="servingH"
          id="serving quantity header"
        >
          <span
            className="hidden"
          >
            Quantity
          </span>
          <i
            alt="serving quantity header"
            className="fas fa-coins"
          />
        </th>
        <td
          className="servingH black"
          headers="serving quantity header"
        >
          1
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="weightH gram"
          id="weight in grams header"
        >
          <span
            className="hidden"
          >
            Serving weight
          </span>
          <i
            alt="weight in grams header"
            className="fas fa-balance-scale"
          />
        </th>
        <td
          className="weightH  gram"
          headers="weight in grams header"
        >
          2
           total gms
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="caloriesH cal calS"
          id="total calories header"
        >
          <span
            className="hidden"
          >
            Total calories
          </span>
          <i
            alt="total calories header"
            className="fas fa-fire-alt"
          />
        </th>
        <td
          className="weightH  gram"
        >
          28
           total cals
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr>
        <th
          className="deleteH"
          id="delete header"
        >
          <span
            className="hidden"
          >
            Delete
          </span>
          <i
            alt="delete header"
            className="fas fa-trash-alt"
          />
        </th>
        <td
          className="delete"
        >
           
          <label
            className="tooltiptext delete"
          >
            click and confirm
            <input
              onClick={[Function]}
              type="radio"
            />
          </label>
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`renders the UI as expected 2`] = `
<section>
  <div
    className="sticky lightBlueBackground"
  >
    <div
      className="sticky"
    >
      <div
        className="iconButtonsContainer"
      >
        <div
          className="back"
        >
          <i
            className="fas fa-arrow-circle-left skin arrowButtons"
            onClick={[Function]}
          />
        </div>
        <div
          className="forward"
        >
          <i
            className="fas fa-arrow-circle-right skin forward arrowButtons"
            onClick={[Function]}
          />
        </div>
      </div>
      <h2
        className="montebello searchResultsTitle white"
      >
         
        See your highest energy results!
      </h2>
    </div>
    <div
      className="iconButtonContainer"
    >
      <button
        className="tooltip iconButtons"
      >
        <i
          className="fas fa-paw black"
          onClick={[Function]}
          type="submit"
        >
          <span
            className="tooltiptext primaryFont white"
          >
            print
          </span>
        </i>
      </button>
      <span
        className="mobileHide"
      >
        <button
          className="tooltip iconButtons"
        >
          <i
            className="fas fa-archive black"
            type="submit"
          >
            <span
              className="tooltiptext primaryFont white"
            >
              save
            </span>
          </i>
        </button>
      </span>
    </div>
    <div
      className="filterButtonContainer moreContainer"
    >
      <a
        aria-current={null}
        href="/add-custom"
        onClick={[Function]}
      >
        <div
          className="montebello red skinBackground moreButton sticky"
        >
          Make your own item
        </div>
      </a>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <div
          className="montebello red skinBackground moreButton sticky"
        >
          Search more items
        </div>
      </a>
    </div>
  </div>
  <table
    className="primaryFont desk"
    id="search-results"
  >
    <thead
      className="blueBackground white"
    >
      <tr>
        <th
          className="imageH tableAdjust"
          id="image header"
        >
          <span
            className="hidden"
          >
            Image
          </span>
        </th>
        <th
          className="itemH"
          id="item name header"
        >
          <span
            className="hidden"
          >
            Name
          </span>
          <i
            alt="item name header"
            className="fas fa-carrot"
          />
        </th>
        <th
          className="brandH"
          id="brand header"
        >
          <span
            className="hidden"
          >
            Brand
          </span>
          <i
            alt="brand header"
            className="fas fa-tag"
          />
        </th>
        <th
          className="servingH"
          id="serving quantity header"
        >
          <span
            className="hidden"
          >
            Quantity
          </span>
          <i
            alt="serving quantity header"
            className="fas fa-coins"
          />
        </th>
        <th
          className="unitH"
          id="unit ie tablespoon or cup header"
        >
          <span
            className="hidden"
          >
            Serving unit
          </span>
          <i
            alt="unit ie tablespoon or cup header"
            className="fas fa-ruler-vertical"
          />
        </th>
        <th
          className="weightH gram"
          id="weight in grams header"
        >
          <span
            className="hidden"
          >
            Serving weight
          </span>
          <i
            alt="weight in grams header"
            className="fas fa-balance-scale"
          />
        </th>
        <th
          className="caloriesH cal calS"
          id="total calories header"
        >
          <span
            className="hidden"
          >
            Total calories
          </span>
          <i
            alt="total calories header"
            className="fas fa-fire-alt"
          />
        </th>
        <th
          className="caloriesH"
          id="total calories per 100grams header"
        >
          <span
            className="hidden"
          >
            Total calories per 100 grams
          </span>
          <i
            alt="total calories per 100grams header"
            className="fab fa-superpowers super"
          />
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</section>
`;
